File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ansMap = {}
        for string in strs:
            key = "".join(sorted(string))
            if key in ansMap:
                ansMap[key].append(string)
            else:
                ansMap[key] = [string]
        ans = []
        for key in ansMap:
            ans.append(ansMap[key])
        return ans

File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const ansMap = new Map();

    for(let i = 0; i < strs.length; i++) {
        let str_sorted = strs[i].split('').sort().join('');
        if (ansMap.has(str_sorted)) {
            ansMap.get(str_sorted).push(strs[i]);
        } else {
            ansMap.set(str_sorted, [strs[i]]);
        }
    } 

    const ans = [];
    for (const [key, value] of ansMap) {
        ans.push(value);
    }
    return ans;
};